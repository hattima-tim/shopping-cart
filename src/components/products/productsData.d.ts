export interface Product {
  path?:string;
  breadCrumbs: Array<{ name: string; path: string }>;
  pathName?: string;
  fullPath: string;
  imagesForProductPage: string[];
  imgForProductCard?: string;
  name: string;
  price: string;
  details: string[];
  fabric: string[];
  color: string[];
  size: string[];
  description: string[];
  additionalInfo: {
    weight: string;
    fabric?: string;
    size: string;
  };
}