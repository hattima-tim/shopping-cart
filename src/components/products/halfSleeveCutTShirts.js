import { Link } from "react-router-dom";
import '../../styles/productsPage.css';
const HalfSleeveCutTShirts = () => {
    return (
        <div className='breadCrumbs'>
           <Link to='/'>Home</Link> / Half Sleeve Cut T-Shirts
        </div>
    )
}

export default HalfSleeveCutTShirts;