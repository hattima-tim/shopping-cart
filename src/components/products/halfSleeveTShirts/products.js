const products = [
  {
    imgForProductCard: "https://res.cloudinary.com/du3oueesv/image/upload/v1659873489/shopping%20cart/half%20sleeve%20cut/mthcssp10-solid-350x435_zvwuma.jpg",
    name: "Half Sleeve Cut and Sew Solid(pattern 15)",
    pathName: "half-sleeve-cut-and-sew-solid-pattern-15",
    price: "৳450",
  },
  {
    imgForProductCard: "https://res.cloudinary.com/du3oueesv/image/upload/v1659873473/shopping%20cart/half%20sleeve%20cut/mthcssp11-solid-350x435_jooe3k.jpg",
    name: "Half Sleeve Cut and Sew Solid(pattern 14)",
    pathName: "half-sleeve-cut-and-sew-solid-pattern-14",
    price: "৳450",
  },
  {
    imgForProductCard: "https://res.cloudinary.com/du3oueesv/image/upload/v1659873453/shopping%20cart/half%20sleeve%20cut/mthcssp12-solid-350x435_zkricr.jpg",
    name: "Half Sleeve Cut and Sew Solid(pattern 13)",
    pathName: "half-sleeve-cut-and-sew-solid-pattern-13",
    price: "৳450",
  },
  {
    imgForProductCard: "https://res.cloudinary.com/du3oueesv/image/upload/v1659873413/shopping%20cart/half%20sleeve%20cut/mthcssp13-solid-350x435_zutbr0.jpg",
    name: "Half Sleeve Cut and Sew Solid(pattern 12)",
    pathName: "half-sleeve-cut-and-sew-solid-pattern-12",
    price: "৳450",
  },
  {
    imgForProductCard: "https://res.cloudinary.com/du3oueesv/image/upload/v1659873393/shopping%20cart/half%20sleeve%20cut/mthcssp14-solid-350x435_sc3l07.jpg",
    name: "Half Sleeve Cut and Sew Solid(pattern 11)",
    pathName: "half-sleeve-cut-and-sew-solid-pattern-11",
    price: "৳450",
  },
  {
    imgForProductCard: "https://res.cloudinary.com/du3oueesv/image/upload/v1659873363/shopping%20cart/half%20sleeve%20cut/mthcssp15-solid-350x435_mz8sws.jpg",
    name: "Half Sleeve Cut and Sew Solid(pattern 10)",
    pathName: "half-sleeve-cut-and-sew-solid-pattern-10",
    price: "৳450",
  },
];

export default products;
