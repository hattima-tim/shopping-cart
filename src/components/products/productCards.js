import { Link } from "react-router-dom";

function ProductCards({ products }) {
  return (
    <div className="productCards">
      {products.map((product, index) => (
        <div className="productCard" key={index}>
          <div className="productCardImg">
            <Link to={`/product/${product.pathName}`}>
              <img src={product.img} alt={product.name} />
            </Link>
          </div>
          <div className="productCard__info">
            <h3>{product.name}</h3>
            <p>{product.price}</p>
          </div>
        </div>
      ))}
    </div>
  );
}

export default ProductCards;
