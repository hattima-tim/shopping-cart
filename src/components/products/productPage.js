import { useState } from "react";
import { Outlet, useParams, Link, useOutletContext } from "react-router-dom";
import uniqid from "uniqid";
import getProductData from "./halfSleeveTShirts/productsData";
import "../../styles/productPage.css";

export function Product() {
  return (
    <>
      <Outlet context={useOutletContext()} />
    </>
  );
}

function ProductPage() {
  const params = useParams();
  const productData = getProductData(params.name);

  const [itemNumber, setItemNumber] = useState(1);
  const itemNumberIncrement = () => {
    setItemNumber(itemNumber + 1);
  };
  const itemNumberDecrement = () => {
    if (itemNumber > 1) {
      setItemNumber(itemNumber - 1);
    }
  };
  const handleChange = (e) => {
    if (e.target.value > 0) {
      const value = parseInt(e.target.value);
      setItemNumber(value);
    }
  };

  const [currentlyShowingAdditionalInfo, setCurrentlyShowingAdditionalInfo] =
    useState("description");
  const showDescription = () => {
    setCurrentlyShowingAdditionalInfo("description");
  };
  const showAdditionalInfo = () => {
    setCurrentlyShowingAdditionalInfo("additionalInfo");
  };

  const [productsInCart, setProductsInCart] = useOutletContext();

  const [fabric, setFabric] = useState('');

  const handleFabricBtnClick = (e) => {
    const currentBtn = e.target;
    const isButtonActive=currentBtn.classList.contains("active");
    
    const fabricBtns = document.querySelectorAll(".fabric-btn");
    fabricBtns.forEach((btn) => {
      if (btn !== currentBtn) {
        btn.classList.remove("active");
      }
    });
    
    if (isButtonActive) {
      currentBtn.classList.remove("active");
    } else {
      currentBtn.classList.add("active");
    }
    setFabric(currentBtn.textContent);
  };

  const [size, setSize] = useState('');

  const handleSizeBtnClick = (e) => {
    const currentBtn = e.target;
    const isButtonActive=currentBtn.classList.contains("active");
    
    const sizeBtns = document.querySelectorAll(".size-btn");
    sizeBtns.forEach((btn) => {
      if (btn !== currentBtn) {
        btn.classList.remove("active");
      }
    });
    
    if (isButtonActive) {
      currentBtn.classList.remove("active");
    } else {
      currentBtn.classList.add("active");
    }
    setSize(currentBtn.textContent);
  };

  let product = {
    name: productData.name,
    price: productData.price,
    quantity: itemNumber,
    totalPrice: itemNumber * (productData.price.split("৳")[1] * 1),
    fabric: fabric,
    size: size,
  };

  const handleAddToCart = () => {
    const newProduct = { ...product };
    setProductsInCart([...productsInCart, newProduct]);
  };

  return (
    <div>
      <div className="main">
        <img
          className="productImage"
          src={productData.img}
          alt={productData.name}
        />
        <div className="mainInfo">
          <div className="breadcrumbs">
            {productData.breadCrumbs.map((breadCrumb, index) => {
              if (index === productData.breadCrumbs.length - 1) {
                return (
                  <Link key={uniqid()} to={breadCrumb.path}>
                    {breadCrumb.name}
                  </Link>
                );
              } else {
                return (
                  <Link key={uniqid()} to={breadCrumb.path}>
                    {breadCrumb.name + "/"}
                  </Link>
                );
              }
            })}
          </div>
          <h1>{productData.name}</h1>
          <h2>{productData.price}</h2>

          <ul className="details">
            {productData.details.map((detail, index) => {
              return <li key={index}>{detail}</li>;
            })}
          </ul>

          <div className="fabricButtons">
            {productData.fabric.map((fabric, index) => {
              return (
                <button
                  key={index}
                  className="productOptionBtn fabric-btn"
                  onClick={handleFabricBtnClick}
                >
                  {fabric}
                </button>
              );
            })}
          </div>

          <div className="sizeButtons">
            {productData.size.map((size, index) => {
              return (
                <button
                  key={index}
                  className="productOptionBtn size-btn"
                  onClick={handleSizeBtnClick}
                >
                  {size}
                </button>
              );
            })}
          </div>

          <div className="addToCart">
            <div className="quantity">
              <button onClick={itemNumberDecrement}>-</button>
              <input type="number" onChange={handleChange} value={itemNumber} />
              <button onClick={itemNumberIncrement}>+</button>
            </div>

            <button className="addToCartButton" onClick={handleAddToCart}>
              Add to Cart
            </button>
          </div>
        </div>
      </div>
      <div className="additional_info">
        <div className="additional_info_tabs">
          <button className="additional_info_tab" onClick={showDescription}>
            Description
          </button>

          <button className="additional_info_tab" onClick={showAdditionalInfo}>
            Additional Information
          </button>
        </div>

        {currentlyShowingAdditionalInfo === "description" && (
          <div className="additional_info_content">
            <p>
              <strong>Size Measurement (in inch/centimeter):</strong>
            </p>

            <ul>
              {productData.description.map((description, index) => {
                return <li key={index}>{description}</li>;
              })}
            </ul>

            <p>
              ডিভাইসের ভিন্নতা অনুযায়ী কালার ডেভিয়েশন এবং ফটোশপ এডিটিং কালার
              ডেভিয়েশনের কারনে বাস্তব ছবি ও সফটকপির মাঝে অনেক সময় পার্থক্য থেকে
              যায়
            </p>
          </div>
        )}
        {currentlyShowingAdditionalInfo === "additionalInfo" && (
          <table className="additional_info_table">
            <tbody>
              <tr>
                <th>Weight</th>
                <td>{productData.additionalInfo.weight}</td>
              </tr>
              <tr>
                <th>Fabric</th>
                <td>{productData.additionalInfo.fabric}</td>
              </tr>
              <tr>
                <th>Size</th>
                <td>{productData.additionalInfo.size}</td>
              </tr>
            </tbody>
          </table>
        )}
      </div>
    </div>
  );
}

export default ProductPage;
