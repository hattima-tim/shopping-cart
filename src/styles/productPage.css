.main{
    display: flex;
    flex-wrap: wrap;
}

.productImage{
    object-fit: cover;
}

.mainInfo{
    flex: 1;
    padding: 0 1rem;
}

.productOptionBtn{
    margin: 0.16rem;
    width: fit-content;
    background: transparent;
    border: 1px solid #A8A8A8;
    border-radius: 3px ;
}

.productOptionBtn:hover{
    border: 2px solid black;
}

