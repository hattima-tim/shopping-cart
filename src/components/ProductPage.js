import { useState } from "react";
import { Outlet, useParams, Link, useOutletContext } from "react-router-dom";
import uniqid from "uniqid";
import ImageMagnifier from "./imageMagnifier";
import "../styles/productPage.css";

export function Product() {
  return (
    <>
      <Outlet context={useOutletContext()} />
    </>
  );
}

function ProductPage({ getProductData }) {
  const params = useParams();
  const productData = getProductData(params.name);

  const [itemNumber, setItemNumber] = useState(1);
  const itemNumberIncrement = () => {
    setItemNumber(itemNumber + 1);
  };
  const itemNumberDecrement = () => {
    if (itemNumber > 1) {
      setItemNumber(itemNumber - 1);
    }
  };
  const handleChange = (e) => {
    if (e.target.value > 0) {
      const value = parseInt(e.target.value);
      setItemNumber(value);
    }
  };

  const [currentlyShowingAdditionalInfo, setCurrentlyShowingAdditionalInfo] =
    useState("description");

  const styleTabAsActive = () => {
    const descriptionTab = document.querySelector(".description");
    descriptionTab.classList.toggle("active-tab");

    const additionalInfoTab = document.querySelector(".additional-info");
    additionalInfoTab.classList.toggle("active-tab");
  };

  const showDescription = () => {
    setCurrentlyShowingAdditionalInfo("description");
    styleTabAsActive();
  };
  const showAdditionalInfo = () => {
    setCurrentlyShowingAdditionalInfo("additionalInfo");
    styleTabAsActive();
  };

  const [productsInCart, setProductsInCart] = useOutletContext();

  const [fabric, setFabric] = useState("");

  const handleFabricBtnClick = (e) => {
    const currentBtn = e.target;
    const isButtonActive = currentBtn.classList.contains("active");

    const fabricBtns = document.querySelectorAll(".fabric-btn");
    fabricBtns.forEach((btn) => {
      if (btn !== currentBtn) {
        btn.classList.remove("active");
      }
    });

    if (isButtonActive) {
      // means that this button is pressed twice to deselect it
      currentBtn.classList.remove("active");
      setFabric("");
    } else {
      currentBtn.classList.add("active");
      setFabric(currentBtn.textContent);
    }
  };

  const [size, setSize] = useState("");

  const handleSizeBtnClick = (e) => {
    const currentBtn = e.target;
    const isButtonActive = currentBtn.classList.contains("active");

    const sizeBtns = document.querySelectorAll(".size-btn");
    sizeBtns.forEach((btn) => {
      if (btn !== currentBtn) {
        btn.classList.remove("active");
      }
    });

    if (isButtonActive) {
      // means that this button is pressed twice to deselect it
      currentBtn.classList.remove("active");
      setSize("");
    } else {
      currentBtn.classList.add("active");
      setSize(currentBtn.textContent);
    }
  };

  let product = {
    id: uniqid(),
    name: productData.name,
    img: productData.img,
    price: productData.price,
    quantity: itemNumber,
    subTotal: itemNumber * (productData.price.split("৳")[1] * 1),
    fabric: fabric,
    size: size,
    path: params.name,
  };

  const handleAddToCart = () => {
    if (size === "" || fabric === "") {
      alert("Please select both fabric and size");
      return;
    }
    const newProduct = { ...product };
    localStorage.setItem(
      "productsInCart",
      JSON.stringify([...productsInCart, newProduct])
    );
    setProductsInCart([...productsInCart, newProduct]);
  };

  return (
    <div className="lg:mx-16">
      <div className="main mx-4 mb-12 mt-4 lg:flex">
        <ImageMagnifier
          src={productData.imgForProductPage}
          alt={productData.name}
        />

        <div className="mainInfo lg:ml-8">
          <div className="breadcrumbs text-sm lg:text-base">
            {productData.breadCrumbs.map((breadCrumb, index) => {
              if (index === productData.breadCrumbs.length - 1) {
                return (
                  <Link key={uniqid()} to={breadCrumb.path}>
                    {breadCrumb.name}
                  </Link>
                );
              } else {
                return (
                  <Link key={uniqid()} to={breadCrumb.path}>
                    {breadCrumb.name + ' / '}
                  </Link>
                );
              }
            })}
          </div>

          <h1 className="mt-2 mb-6 text-xl font-bold text-gray-600 lg:text-3xl">
            {productData.name}
          </h1>
          <h2 className="mb-4 text-2xl font-bold text-slate-800">
            {productData.price}
          </h2>
          <p className="mb-6">
            <strong>Detailed Specification</strong>
          </p>
          <ul className="details list-disc">
            {productData.details.map((detail, index) => {
              return (
                <li key={index} className="mx-4 my-3">
                  {detail}
                </li>
              );
            })}
          </ul>

          {productData.fabric.toString() !== "" && ( // for some products fabric data is not present
            <div className="fabricButtons">
              <span className="text-sm font-bold text-slate-800">FABRIC</span>
              <br />
              {productData.fabric.map((fabric, index) => {
                return (
                  <button
                    key={index}
                    className="fabric-btn mb-1 mr-2 w-fit rounded-sm border-2 border-black bg-white p-1 text-base text-black"
                    onClick={handleFabricBtnClick}
                  >
                    {fabric}
                  </button>
                );
              })}
            </div>
          )}

          <div className="sizeButtons">
            <span className="text-sm font-bold text-slate-800">SIZE</span>
            <br />
            {productData.size.map((size, index) => {
              return (
                <button
                  key={index}
                  className="size-btn mb-8 mr-2 w-fit rounded-sm border-2 border-black bg-white p-1 text-base text-black"
                  onClick={handleSizeBtnClick}
                >
                  {size}
                </button>
              );
            })}
          </div>

          <div className="addToCart flex gap-4">
            <div className="quantity">
              <button
                onClick={itemNumberDecrement}
                className="h-10 w-8 bg-gray-200 hover:bg-slate-300"
              >
                -
              </button>
              <input
                type="number"
                onChange={handleChange}
                value={itemNumber}
                className="h-10 w-8 border border-y-gray-200 text-center"
              />
              <button
                onClick={itemNumberIncrement}
                className="h-10 w-8 bg-gray-200 hover:bg-slate-300"
              >
                +
              </button>
            </div>

            <button
              onClick={handleAddToCart}
              className="w-fit bg-black px-6 py-2 font-bold text-white "
            >
              Add to Cart
            </button>
          </div>
        </div>
      </div>
      <div className="additional_info m-4">
        <div className="additional_info_tabs md:border-top-gray-400 md:flex md:gap-2 md:border-t">
          <button
            onClick={showDescription}
            className="description active-tab peer border-t py-2 text-start font-bold hover:border-t-2 hover:border-t-black peer-hover:border-t peer-hover:border-t-gray-400"
          >
            Description
          </button>
          <br />
          <button
            onClick={showAdditionalInfo}
            className="additional-info peer border-t py-2 text-start font-bold hover:border-t-2 hover:border-t-black "
          >
            Additional Information
          </button>
        </div>

        {currentlyShowingAdditionalInfo === "description" && (
          <div className="additional_info_content my-4">
            <p>
              <strong>Size Measurement (in inch/centimeter):</strong>
            </p>

            <ul>
              {productData.description.map((description, index) => {
                return (
                  <li key={index} className="my-4">
                    {description}
                  </li>
                );
              })}
            </ul>

            <p className="my-4">
              ডিভাইসের ভিন্নতা অনুযায়ী কালার ডেভিয়েশন এবং ফটোশপ এডিটিং কালার
              ডেভিয়েশনের কারনে বাস্তব ছবি ও সফটকপির মাঝে অনেক সময় পার্থক্য থেকে
              যায়
            </p>
          </div>
        )}
        {currentlyShowingAdditionalInfo === "additionalInfo" && (
          <table className="additional_info_table my-4">
            <tbody>
              <tr>
                <th>Weight</th>
                <td>{productData.additionalInfo.weight}</td>
              </tr>
              <tr>
                <th>Fabric</th>
                <td>{productData.additionalInfo.fabric}</td>
              </tr>
              <tr>
                <th>Size</th>
                <td>{productData.additionalInfo.size}</td>
              </tr>
            </tbody>
          </table>
        )}
      </div>
    </div>
  );
}

export default ProductPage;
