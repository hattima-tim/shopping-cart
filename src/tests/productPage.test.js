import { render, screen } from "@testing-library/react";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import { Product } from "../components/products/productPage";
import ProductPage from "../components/products/productPage";
import Header from "../components/header";

const setup = () => {
  const { container } = render(
    <MemoryRouter
      initialEntries={["/product/half-sleeve-cut-and-sew-solid-pattern-15"]}
    >
      <Routes>
        <Route path="/" element={<Header />}>
          <Route path="product" element={<Product />}>
            <Route path=":name" element={<ProductPage />} />
          </Route>
        </Route>
      </Routes>
    </MemoryRouter>
  );
  return { container };
};

test("Product page", () => {
  const { container } = setup();
  expect(container).toMatchSnapshot();
});

describe("item counter", () => {
  test("increment and decrement", () => {
    setup();

    const inputField = screen.getByDisplayValue("1");
    const decrementBtn = screen.getByRole("button", { name: "-" });
    const incrementBtn = screen.getByRole("button", { name: "+" });

    userEvent.click(incrementBtn);
    userEvent.click(incrementBtn);
    userEvent.click(incrementBtn);

    userEvent.click(decrementBtn);

    expect(inputField.value).toBe("3");
  });

  test("user can type in input field", () => {
    setup();

    const inputField = screen.getByDisplayValue("1");

    userEvent.clear(inputField);
    userEvent.type(inputField, "56");
    expect(inputField.value).toBe("56");
  });
});
