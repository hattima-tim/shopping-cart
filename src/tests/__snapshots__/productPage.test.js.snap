// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product page 1`] = `
<div>
  <div
    class="header"
  >
    <img
      alt="one ummah logo"
      class="companyLogo"
      src="https://res.cloudinary.com/du3oueesv/image/upload/v1660122044/shopping%20cart/oubd-logo-with-tm-black-1-1400x642_se0t70.png"
    />
    <div
      class="shopping-cart-icon"
    >
      <span
        class="price"
      >
        ৳ 
        0
      </span>
      <span
        class="fa-layers fa-fw"
      >
        <svg
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 0C107.5 0 117.4 8.19 119.6 19.51L121.1 32H541.8C562.1 32 578.3 52.25 572.6 72.66L518.6 264.7C514.7 278.5 502.1 288 487.8 288H170.7L179.9 336H488C501.3 336 512 346.7 512 360C512 373.3 501.3 384 488 384H159.1C148.5 384 138.6 375.8 136.4 364.5L76.14 48H24C10.75 48 0 37.25 0 24C0 10.75 10.75 0 24 0H96zM128 464C128 437.5 149.5 416 176 416C202.5 416 224 437.5 224 464C224 490.5 202.5 512 176 512C149.5 512 128 490.5 128 464zM512 464C512 490.5 490.5 512 464 512C437.5 512 416 490.5 416 464C416 437.5 437.5 416 464 416C490.5 416 512 437.5 512 464z"
          />
        </svg>
        <span
          class="fa-layers-counter"
          data-testid="cart-item-count"
          style="background: Tomato;"
        >
          0
        </span>
      </span>
      <div
        class="products_tooltip"
      >
        <p
          class="empty_cart"
        >
          Your cart is empty
        </p>
      </div>
    </div>
  </div>
  <div>
    <div
      class="main"
    >
      <img
        alt="Half Sleeve Cut and Sew Solid(pattern 15)"
        class="productImage"
        src="https://res.cloudinary.com/du3oueesv/image/upload/v1659873489/shopping%20cart/half%20sleeve%20cut/mthcssp10-solid-350x435_zvwuma.jpg"
      />
      <div
        class="mainInfo"
      >
        <div
          class="breadcrumbs"
        >
          <a
            href="/"
          >
            HOME/
          </a>
          <a
            href="/half-sleeve-cut-and-sew-solid"
          >
            HALF SLEEVE CUT T-SHIRTS
          </a>
        </div>
        <h1>
          Half Sleeve Cut and Sew Solid(pattern 15)
        </h1>
        <h2>
          ৳450
        </h2>
        <ul
          class="details"
        >
          <li>
            Fabric: 100% Cotton
          </li>
          <li>
            GSM: 160-165
          </li>
        </ul>
        <div
          class="fabricButtons"
        >
          <button
            class="productOptionBtn fabric-btn"
          >
            COMBED COTTON
          </button>
        </div>
        <div
          class="sizeButtons"
        >
          <button
            class="productOptionBtn size-btn"
          >
            M
          </button>
          <button
            class="productOptionBtn size-btn"
          >
            L
          </button>
          <button
            class="productOptionBtn size-btn"
          >
            XL
          </button>
          <button
            class="productOptionBtn size-btn"
          >
            XXL
          </button>
        </div>
        <div
          class="addToCart"
        >
          <div
            class="quantity"
          >
            <button>
              -
            </button>
            <input
              type="number"
              value="1"
            />
            <button>
              +
            </button>
          </div>
          <button
            class="addToCartButton"
          >
            Add to Cart
          </button>
        </div>
      </div>
    </div>
    <div
      class="additional_info"
    >
      <div
        class="additional_info_tabs"
      >
        <button
          class="additional_info_tab"
        >
          Description
        </button>
        <button
          class="additional_info_tab"
        >
          Additional Information
        </button>
      </div>
      <div
        class="additional_info_content"
      >
        <p>
          <strong>
            Size Measurement (in inch/centimeter):
          </strong>
        </p>
        <ul>
          <li>
            M(Medium): Chest 39.37″ inch/ 100 cm, Height 27.55″ inch/ 70 cm
          </li>
          <li>
            L(large) : Chest 40.94″inch/ 104 cm, Height 28.34″inch/ 72 cm
          </li>
          <li>
            XL(Extra large): Chest 42.51″ inch/ 108 cm, Height 29.13″ inch/ 74 cm
          </li>
          <li>
            XXL(Extra large) : Chest 44.09″ inch/ 112 cm, Height 29.92″ inch/ 76 cm
          </li>
        </ul>
        <p>
          ডিভাইসের ভিন্নতা অনুযায়ী কালার ডেভিয়েশন এবং ফটোশপ এডিটিং কালার ডেভিয়েশনের কারনে বাস্তব ছবি ও সফটকপির মাঝে অনেক সময় পার্থক্য থেকে যায়
        </p>
      </div>
    </div>
  </div>
</div>
`;
